src/auth/User.h

```diff
@@ -48,7 +51,8 @@ class User : public RefCountable
     Auth::Type auth_type;
     /** the config for this user */
     Auth::SchemeConfig *config;
-    dlink_list proxy_match_cache;
+    // XXX would be better served by unordered_map<void*, int> ?
+    std::list<acl_proxy_auth_match_cache> proxy_match_cache;
```

Comment:
Given that @yadij is planning to change this code further to LruMap, I'd leave as-is with minimal changes

Reply:
> Given that @yadij is planning to change this code further to LruMap, I'd leave as-is with minimal changes

If "leave as-is with changes" means "an std::list class with a PoolingAllocator" (i.e. the first out of three options in my change request), then I (still) support that plan.

<!-- Otherwise, please keep in mind that those three options are the only viable options for going forward that I know about. Amos' long-term plans do not justify introducing a performance regression in _this_ PR and "leaving as is" would introduce such a regression (by removing pooling of memory allocations, as discussed). -->

Reply:
> > Given that @yadij is planning to change this code further to LruMap, I'd leave as-is with minimal changes
> 
> If "leave as-is with changes" means "an std::list class with a PoolingAllocator" (i.e. the first out of three options in my change request), then I (still) support that plan.

yes, that's what it means. In the end this class is a map, so unless these caches are super small (and therefore of little benefit) using a real map is easier to read and (probably) faster as well